homeassistant/components/onvif/parsers.py

```diff
@@ -10,6 +11,15 @@
 PARSERS: Registry[str, Callable[[str, Any], Coroutine[Any, Any, Event]]] = Registry()
 
 
+def datetime_or_zero(value: str):
+    """Convert strings to datetimes, if invalid, return datetime.min."""
+    # To handle cameras that return times like '0000-00-00T00:00:00Z' (e.g. hikvision)
+    try:
+        return dt_util.parse_datetime(value)
```

Comment:
`parse_datetime` may return `None`. We should probably guard for that somehow.

Reply:
Done